crates/biome_lsp/src/capabilities.rs

```diff
@@ -19,12 +40,24 @@ pub(crate) fn server_capabilities(capabilities: &ClientCapabilities) -> ServerCa
         text_document_sync: Some(TextDocumentSyncCapability::Kind(
             TextDocumentSyncKind::INCREMENTAL,
         )),
-        document_formatting_provider: Some(OneOf::Left(true)),
-        document_range_formatting_provider: Some(OneOf::Left(true)),
-        document_on_type_formatting_provider: Some(DocumentOnTypeFormattingOptions {
-            first_trigger_character: String::from("}"),
-            more_trigger_character: Some(vec![String::from("]"), String::from(")")]),
-        }),
+        document_formatting_provider: if supports_formatter_dynamic_registration {
+            None
+        } else {
+            Some(OneOf::Left(true))
+        },
```

Comment:
If you can thing of a more elegant way to write this, I'm all ears!

Reply:
This `supports_formatter_dynamic_registration` comes from a chain of `Option`; we can remove the `unwrap_or`, and change it to a `map`:

```rs
    let supports_on_type_formatter_dynamic_registration = capabilities
        .text_document
        .as_ref()
        .and_then(|text_document| text_document.on_type_formatting.as_ref())
        .and_then(|on_type_formatting| on_type_formatting.dynamic_registration)
		.map(|supports_it| {
			if supports_it {
				None
			} else {
				Some(OneOf::Left(true))
			}
		});

```

Reply:
Great suggestion, thanks!

Reply:
Actually, map didnt seem to work.

I went with the following instead.

```rust
.and_then(|supported| {
	if supported {
		None
	} else {
		Some(OneOf::Left(true))
	}
})
```

Reply:
Yeah I always make a mistake between them ahah